apps/mull-ui/src/app/pages/messages/announcements/announcements.tsx

```diff
@@ -1,16 +1,110 @@
-import React from 'react';
-import { mockPosts } from '../../../../constants';
+import { IChatForm, ISerializedPost } from '@mull/types';
+import { useFormik } from 'formik';
+import { History } from 'history';
+import React, { useContext } from 'react';
+import { Redirect } from 'react-router-dom';
+import { toast } from 'react-toastify';
+import * as Yup from 'yup';
+import { ROUTES } from '../../../../../src/constants';
+import {
+  Event,
+  PostAddedDocument,
+  useChannelByEventQuery,
+  useCreatePostMutation,
+} from '../../../../generated/graphql';
 import { ChatInput } from '../../../components';
 import ChatBubbleList from '../../../components/chat-bubble-list/chat-bubble-list';
+import UserContext from '../../../context/user.context';
+import { useToast } from '../../../hooks/useToast';
 
-export const AnnouncementsPage = () => {
-  // TODO: Replace the following with the real announcement page implementation
+interface subscriptionData {
+  subscriptionData: {
+    data: {
+      postAdded: Partial<ISerializedPost>;
+    };
+  };
+}
 
-  return (
-    <div className="announcement-page">
-      <ChatBubbleList posts={mockPosts} />
-      <ChatInput />
-    </div>
-  );
+export interface AnnouncementPageProps {
+  history: History;
+}
+
+export const AnnouncementsPage = ({ history }: AnnouncementPageProps) => {
+  const { updateToast } = useToast();
+  const [createPostMutation] = useCreatePostMutation();
+  const { userId } = useContext(UserContext);
+  const { data, loading, error, subscribeToMore } = useChannelByEventQuery({
+    variables: {
+      eventId: 1, //TODO: Replace with dynamic event ID
+      channelName: 'Announcements',
+    },
+  });
+
+  const isEventHost = (event: Event, currentUserId: number) => {
```

Comment:
I think this page has the exact same logic as the one for group chat, except for the permission part for the chat input and the channel name. Do you it is a good idea to extract the common code somewhere so that it can be reused for group chat?

Reply:
Good idea, done!